src/frontend/src/catalog/root_catalog.rs

```diff
@@ -197,7 +197,10 @@ impl Catalog {
 
     pub fn drop_database(&mut self, db_id: DatabaseId) {
         let name = self.db_name_by_id.remove(&db_id).unwrap();
-        let _database = self.database_by_name.remove(&name).unwrap();
+        let database = self.database_by_name.remove(&name).unwrap();
+        database.iter_all_table_ids().for_each(|table| {
+            self.table_by_id.remove(&table);
```

Comment:
Why do we need a field to store all tables in the `RootCatalog`? 🥺 It seems really error-prone if we get it mismatched with the one stored in `SchemaCatalog`. At least using `Arc` can be better here.

Reply:
Yes, it was introduced a long time ago for internal table function `rw_table` and later used to get vnode mappings in fragmenter. I tried to deprecate it but it seems not that easy. 🥵 Let's change it to use `Arc` for now.